src/VisualStudio/Core/Def/DesignerAttribute/VisualStudioDesignerAttributeService.cs

```diff
@@ -125,10 +124,25 @@ private async ValueTask ProcessWorkspaceChangeAsync(CancellationToken cancellati
             var trackingService = _workspace.Services.GetRequiredService<IDocumentTrackingService>();
             var priorityDocumentId = trackingService.GetActiveDocument(solution)?.Id;
 
+            // If there is an active document, then process changes to it right away, so that the UI updates quickly
+            // when the user adds/removes a form from a particular document.
+            if (priorityDocumentId != null)
+            {
+                await client.TryInvokeAsync<IRemoteDesignerAttributeDiscoveryService>(
+                    solution,
+                    (service, checksum, callbackId, cancellationToken) => service.DiscoverDesignerAttributesAsync(
+                        callbackId, checksum, priorityDocumentId, cancellationToken),
+                    callbackTarget: this,
+                    cancellationToken).ConfigureAwait(false);
+            }
+
+            // Wait a little after the priority document and process the rest of the solution at a lower priority.
+            await _listener.Delay(DelayTimeSpan.NonFocus, cancellationToken).ConfigureAwait(false);
+
             await client.TryInvokeAsync<IRemoteDesignerAttributeDiscoveryService>(
                 solution,
                 (service, checksum, callbackId, cancellationToken) => service.DiscoverDesignerAttributesAsync(
-                    callbackId, checksum, priorityDocumentId, cancellationToken),
+                    callbackId, checksum, cancellationToken),
                 callbackTarget: this,
                 cancellationToken).ConfigureAwait(false);
         }
```

Comment:
Is there any concern this might cause more resyncs to OOP if the solution moved forward between the earlier use and here?

Reply:
yes.  but i'm realllllly not bothered by that. we already have tons of features that wake up and are making all those calls.

Practically, because of the delay above, we only do this work anyways when nothing has been happening and we're just idle.  so this is ok to me.